packages/financial-templates-lib/src/logger/PagerDutyV2Transport.ts

```diff
@@ -0,0 +1,52 @@
+// This transport enables winston logging to send messages to pager duty v2 api.
+import Transport from "winston-transport";
+import { event } from "@pagerduty/pdjs";
+
+type TransportOptions = ConstructorParameters<typeof Transport>[0];
+export type Severity = "critical" | "error" | "warning" | "info";
+
+interface Config {
+  integrationKey: string;
+  customServices?: { [key: string]: string };
+}
+
+export class PagerDutyV2Transport extends Transport {
+  private readonly integrationKey: string;
+  private readonly customServices: { [key: string]: string };
+  constructor(winstonOpts: TransportOptions, { integrationKey, customServices = {} }: Config) {
+    super(winstonOpts);
+    this.integrationKey = integrationKey;
+    this.customServices = customServices;
+  }
+  // pd v2 severity only supports critical, error, warning or info.
+  public static convertLevelToSeverity(level?: string): Severity {
+    if (!level) return "error";
+    if (level === "warn") return "warning";
+    if (level === "info" || level === "critical") return level;
+    return "error";
+  }
+  // Note: info must be any because that's what the base class uses.
+  async log(info: any, callback: () => void): Promise<void> {
+    try {
+      // we route to different pd services using the integration key (routing_key), or multiple services with the custom services object
```

Comment:
your comment here says "multiple services with the custom services object". just to be clear: the service key that you define here will do the multiple routing, right? as in you send it to _one_  `routing_key` and then pagerduty sends it to multiple escalation paths/servicesIds?

Reply:
hmm, the way i understand this ( and i could be wrong), this routing_key does double duty as your api key as well as your destination service.  As far as i know its only 1 routing key per service (not one to many), but you can essentially configure a single logger to emit to multiple services by setting the `customServices[info.notificationPath]` with each of their routing keys.